CHANGELOG.md

```diff
@@ -2,11 +2,79 @@
 
 ## 0.11.0-dev (master)
 
+### Deprecations
+
+0.11 is last major release planned before we merge the `unifiprotect` integration into core. As a result, a number of features are being removed when we merged into core.
+
+The following services will be removed in the next version:
+
+* `unifiprotect.set_recording_mode` -- use the select introduced in 0.10 instead
+* `unifiprotect.set_ir_mode` -- use the select entity introduced in 0.10 instead
+* `unifiprotect.set_status_light` -- use the switch entity on the camera device instead
+* `unifiprotect.set_hdr_mode` -- use the switch entity on the camera device instead
+* `unifiprotect.set_highfps_video_mode` -- use the switch entity on the camera device instead
+* `unifiprotect.set_doorbell_lcd_message` -- use the select entity introduced in 0.10 instead
+* `unifiprotect.set_mic_volume` -- use the number entity introduced in 0.10 instead
+* `unifiprotect.set_privacy_mode` -- use the switch entity introduced in 0.10 instead
+* `unifiprotect.set_zoom_position` -- use the number entity introduced in 0.10 instead
+* `unifiprotect.set_wdr_value` -- use the number entity introduced in 0.10 instead
+* `unifiprotect.light_settings` -- use the select entity introduced in 0.10 instead
+* `unifiprotect.set_viewport_view` -- use the select entity introduced in 0.10 instead
+
+The following events will be removed in the next version:
+
+* `unifiprotect_doorbell` -- use a State Changed event on "Doorbell" binary sensor on the device instead
+* `unifiprotect_motion` -- use a State Changed event on the "Motion" binary sensor on the device instead
+
+All of following attributes should be duplicated data that can be gotten from other devices/entities and as such, they will be removed in the next version.
+
+* `device_model` will be removed from all entities
+* `last_tripped_time` will be removed from binary sensor entities
+* `up_since` will be removed from camera and light entities
+* `enabled_at` will be removed from light entities
+* `camera_id`, `chime_duration`, `is_dark`, `mic_sensitivity`, `privacy_mode`, `wdr_value`, and `zoom_position`  will be removed from camera entities
```

Comment:
Make sure you guys both review the changelog. Especially the deprecations section. That is everything I was thinking that we were remove for when we merge to core based on @bdraco's list of TODOs.

last_tripped_time is one of them. I was thinking that it was redundant since HA would have a record of the last time the second was active.

Reply:
The services were initially created to make it possible to set some values from an automation, which was not really possible from any of the entities. With the addition of the Select and Number entity plus a few new Switch entities, I agree that these can be removed, as all of the Services now can be executed from an automation using the built-in entity services.

Of the attributes, it is only the last_tripped_time I am not sure about can be removed, as this was specifically requested by users many releases ago. The last_changed time in HA, just indicates when it changed state the last time, and not when it came on - as far as I know?

My five cents.

Reply:
> The last_changed time in HA, just indicates when it changed state the last time, and not when it came on - as far as I know?

That is what I was unsure about as well. That is why I added it to the list, so we could talk about it. It is a binary sensor, so I would assume the last changed time _was_ the last time it was tripped.

Reply:
I also just added the "Motion Recording" sensors (cameras) and "Light Turn On" sensors (lights) to this list as well since the selects that exist are even better.

Reply:
> so I would assume the last changed time was the last time it was tripped.

I will do a quick test on that.

Reply:
Test completed. The last_changed shows when the state was last changed. So going from on to off, resets the last_changed value. Therefore I think we should keep the last_tripped_time.

Reply:
From my testing, they almost lined up perfectly. 

![image](https://user-images.githubusercontent.com/490848/143879547-9b20ef03-a1d9-42b0-9d5c-dbcbe8098b48.png)

Reply:
Unfortunately, as the current code for how it works stands, that is how `last_tripped_time` works now as well. The `last_motion` attribute that comes from UniFi Protect is constantly updated until the motion event ends. So `last_motion` == `last_changed` within a reasonably degree of accurately.

Reply:
Added logging to test/verify this:

```
2021-11-29 14:22:25 DEBUG (MainThread) [pyunifiprotect.api] Skipping update since websocket is active
2021-11-29 14:22:27 DEBUG (MainThread) [custom_components.unifiprotect.data] {'last_motion': datetime.datetime(2021, 11, 29, 14, 22, 27, 235000, tzinfo=datetime.timezone.utc), 'is_motion_detected': False, 'is_recording': True}
2021-11-29 14:22:27 DEBUG (MainThread) [custom_components.unifiprotect.data] {'last_motion': datetime.datetime(2021, 11, 29, 14, 22, 35, 568000, tzinfo=datetime.timezone.utc)}
2021-11-29 14:22:27 DEBUG (MainThread) [pyunifiprotect.api] Skipping update since websocket is active
2021-11-29 14:22:29 DEBUG (MainThread) [pyunifiprotect.api] Skipping update since websocket is active
2021-11-29 14:22:31 DEBUG (MainThread) [custom_components.unifiprotect.data] {'last_motion': datetime.datetime(2021, 11, 29, 14, 22, 31, 669000, tzinfo=datetime.timezone.utc), 'is_motion_detected': False, 'is_recording': True}
2021-11-29 14:22:31 DEBUG (MainThread) [pyunifiprotect.api] Skipping update since websocket is active
2021-11-29 14:22:31 DEBUG (MainThread) [custom_components.unifiprotect.data] {'last_motion': datetime.datetime(2021, 11, 29, 14, 22, 39, 668000, tzinfo=datetime.timezone.utc)}
```

When `is_motion_detected: False` is _not_ passed, that is a message to update the `last_motion` timestamp because motion is ongoing.

Reply:
OK, I get this. The still remaining issue is that if people use this attribute in automations, it is much easier from a user perspective to check for `last_tripped_time` than to figure out how to get the last updated. So we should at least document that in the release notes.

Reply:
I am updating the attributes in the changelog for where to get the data from instead. 

* `device_model`: Might be more for @bdraco, but how do I get data about the entity's device? The model is on the device itself, but how do you get that for an automation? I now the devices are more of an abstract, but that data is still there in the `entity_description`
* `last_tripped_time` -> `last_changed`
* `up_since` -> Is there a way to get how long it has been since an entity was unavailable? Otherwise, should we keep this one?
* `camera_id`: Looking at it, we do not actually store the device ID on the device/`entity_description`, so maybe we should keep this one? Still deprecate it and move it to `device_id` on _on_ entities?

Reply:
Many of these attributes were added in the early days of the Integration, and were more used by me to debug stuff, than for actual practical use. So in my book, only last_tripped_time needs to be documented. The rest can just be deprecated. I will be surprised if anyone ask for them back.

Reply:
Updates the defaults on each of them and added a new "Uptime" sensor to replace the `up_since` attribute.